src/C50A0.c

```diff
@@ -116,13 +123,40 @@ void item_entity_enable_shadow(ItemEntity* itemEntity) {
     }
 }
 
-INCLUDE_ASM(s32, "C50A0", clear_item_entity_data);
+void clear_item_entity_data(void) {
+    s32 i;
+
+    if (!gGameStatusPtr->isBattle) {
+        D_801565A0 = D_80155DA0;
+    } else {
+        D_801565A0 = D_801561A0;
+    }
+
+    for (i = ARRAY_COUNT(D_80155DA0); i >= 0; i--) {
```

Comment:
This will match as a count up; it's a GCC optimisation for simple loop bodies
```suggestion
    for (i = 0; i < ARRAY_COUNT(D_80155DA0); i++) {
```